torch/_inductor/codegen/cuda/cuda_kernel.py

```diff
@@ -265,21 +263,196 @@ def row_or_column_stride(self, node: IRNode, default_value: int = 0) -> str:
             )
 
 
+class CUDATemplateBuffer(TemplateBuffer):
+    def __init__(
+        self,
+        template: "CUDATemplate",  # type: ignore[name-defined]
```

Comment:
It's neccessary to expose the template and the op that is being wrapped, in order to make a valid decision on whether a Node may be (EVT-) fused or not. As this information was previously hidden in inaccessible closures, I needed to refactor a bit.